src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -748,3 +751,77 @@ impl RustcTargetData {
         self.target_config(kind).links_overrides.get(lib_name)
     }
 }
+
+/// Structure used to deal with Rustdoc fingerprinting
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RustDocFingerprint {
+    pub rustc_vv: String,
```

Comment:
Because the docfiles only depend on the compiler version, if it changes we always rebuild deleting the entire docs folder. Otherways, if the compiler version is the same, we know 100% sure that the docfiles are still valid and will not introduce any errors on the browser console or malfunctions.

We simply don't need anymore data than the version, that's why we don't use the full fingerprint of the build.

Reply:
What about if people change the feature they are building with, which causes different items to be built? Also is docs only depending on compiler version and no flags something that we guarantee?

Reply:
If you check the issue from which the PR was originated. The goal is to ensure that the html, css and js files are consistent with the compiler version currently used and not mix it up. 

Any other thing stills being handled on the same way it was before. This has nothing to do with the content that you're documenting ie. target, features etc. But with the actual js, css and html code in the files of the docs and which version should they be used on.

Hope that makes it a bit more clear.

Reply:
@guswynn Each crate has a "fingerprint" stored in the `.fingerprint` directory. The things it tracks are listed [here](https://github.com/rust-lang/cargo/blob/ccf781ab902d53ab80d76aa0e34f91df0d49e4b3/src/cargo/core/compiler/fingerprint.rs#L56-L77).  The fingerprint will detect if an individual crate needs to be rebuilt, but it doesn't detect when shared files (like the search index) need to be deleted and rebuilt.  This adds a "global" fingerprint for building docs to detect when the rust toolchain changes (and may introduce an incompatible search index), and clears out the entire docs directory to deal with those changes.  Changes to things like features only trigger the individual crate to be rebuilt, but it retains the shared files (which rustdoc updates incrementally).